addons/im_livechat/static/src/embed/common/livechat_button.js

```diff
@@ -43,6 +43,17 @@ export class LivechatButton extends Component {
             },
         });
         useExternalListener(document.body, "scroll", this._onScroll, { capture: true });
+        useEffect(
+            () => {
+                const rootNodeClassList = this.ref.el?.getRootNode().host?.classList;
+                if (rootNodeClassList) {
+                    this.isShown
+                        ? rootNodeClassList.add("o-livechat-LivechatButton-isVisible")
+                        : rootNodeClassList.remove("o-livechat-LivechatButton-isVisible");
+                }
+            },
+            () => [this.isShown]
+        );
```

Comment:
Should destroy `o-livechat-LivechatButton-isVisible` on unmount.

_(Even though I don't think we have such a use-case, it feels wrong to not clean classname)_

Reply:
I guess it can't hurt, since we modify html out of the component itself, we better handle every case.

Reply:
Shouldn't it be done with the return callback of useEffect instead?

Reply:
What callback are you talking about?

Reply:
`useEffect` declares a return value `void`, the documentation mentions some cleanup function to be run before patching/unmounting but the function definition doesn't explain how it works.
So you have outdated information, the documentation is outdated or I'm missing something.

![image](https://github.com/user-attachments/assets/635b8002-5e1a-4729-a0f8-adf1f2584bf5)

Reply:
It's `void` because it is not the return value of the `useEffect`, but the return value of the callback param (= `Effect`).

So you have to check the doc of `Effect`.

Reply:
So I should do something like this.

```js
        useEffect(
            () => {
                const rootNodeClassList = this.ref.el?.getRootNode().host?.classList;
                if (rootNodeClassList) {
                    this.isShown
                        ? rootNodeClassList.add("o-livechat-LivechatButton-isVisible")
                        : rootNodeClassList.remove("o-livechat-LivechatButton-isVisible");
                }
                return () => rootNodeClassList.remove("o-livechat-LivechatButton-isVisible");
            },
            () => [this.isShown]
        );
```

Reply:
The fact that Patch is also going to trigger the function isn't an issue ?

Reply:
It's actually beneficial, I believe with the cleanup return we no longer need to handle the `this.isShown` "else" case as the cleanup would be ran when `this.isShown` changes, and a simple "if" on `this.isShown` would not put it back.